src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * A class that deals with the interaction of the user.
+ *
+ * @author Safwan A0235287X
+ */
+public class Parser {
+
+    private TaskList taskList;
+
+    /**
+     * Constructor to create Parser object.
+     * @param taskList the input list for the storage of data.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Method to read and process input given by the user.
+     * @throws DukeException
+     * @throws IOException
+     */
+    public void readInput() throws DukeException, IOException {
+        while (true) {
+
+            Scanner input = new Scanner(System.in);
+            String s = input.nextLine();
+
+            if (s.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon.");
+                break;
+            }
+
+            String[] word = s.split(" ");
+
+            if (s.equals("list")) {
+                taskList.list();
+
+            } else if (word[0].equals("mark")) {
+                Integer num = Integer.parseInt(word[1]);
+                taskList.mark(num);
+
+            } else if (word[0].equals("unmark")) {
+                Integer num = Integer.parseInt(word[1]);
+                taskList.unmark(num);
+
+            } else if (word[0].equals("todo")) {
+                if (s.endsWith("todo")) {
+                    throw new DukeException("Ooops, the description of todo cannot be empty!");
+                }
+
+                String substringtd = s.replaceAll("todo ", "");
```

Comment:
Instead of naming variable as `substringtd`, maybe consider renaming it to something likes `substringTd` or `todoSubstring`? According to our coding standard:

> Variable names must be in **camelCase**.